prboom2/src/gl_main.c

```diff
@@ -2253,7 +2272,23 @@ static void gld_DrawSprite(GLSprite *sprite)
     {
       glGetIntegerv(GL_BLEND_SRC, &blend_src);
       glGetIntegerv(GL_BLEND_DST, &blend_dst);
-      glBlendFunc(GL_DST_COLOR, GL_ONE_MINUS_SRC_ALPHA);
+      switch (gl_thingspritefuzzmode)
+      {
+          case fuzz_darken:
+              glBlendFunc(GL_DST_COLOR, GL_ONE_MINUS_SRC_ALPHA);
+              break;
+          case fuzz_shadow:
+              glBlendFunc(GL_ZERO, GL_ONE_MINUS_SRC_ALPHA);
+              break;
+          case fuzz_transparent:
+              glBlendFunc(GL_ONE, GL_ONE_MINUS_SRC_ALPHA);
+              break;
+          default:
```

Comment:
I don't think this is possible to happen. Config values should get normalized upon loading the config file and it's impossible to set them to invalid values in-game. To silence the compiler I would just merge thd default rule and the fuzz_darken case.

Reply:
Fair enough, I was being overprotective in case of future bugs. But merging is a reasonable option too so I will implement that.